SEGV when specify scheme name ahead of table name in create table

qbguan@qbguan:~$ echo "create table hello.t (id int unsigned);" | soar
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x92d417]

goroutine 1 [running]:
github.com/XiaoMi/soar/ast.GetMeta.func1(0xbc5ee0, 0xc420186240, 0xaeb4c0, 0xc4203c0ee8, 0x410078)
	/home/zhangliang3/workspace/go/src/github.com/XiaoMi/soar/ast/meta.go:98 +0xd47
github.com/XiaoMi/soar/vendor/vitess.io/vitess/go/vt/sqlparser.Walk(0xc42057fd80, 0xc4203c0f40, 0x1, 0x1, 0xc420186240, 0xc4203c0f50)
	/home/zhangliang3/workspace/go/src/github.com/XiaoMi/soar/vendor/vitess.io/vitess/go/vt/sqlparser/ast.go:239 +0x72
github.com/XiaoMi/soar/ast.GetMeta(0xbcac00, 0xc420186240, 0x0, 0x2)
	/home/zhangliang3/workspace/go/src/github.com/XiaoMi/soar/ast/meta.go:76 +0x103
github.com/XiaoMi/soar/advisor.(*Query4Audit).RuleMultiDBJoin(0xc4202aa880, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
	/home/zhangliang3/workspace/go/src/github.com/XiaoMi/soar/advisor/heuristic.go:1986 +0x12d
main.main()
	/home/zhangliang3/workspace/go/src/github.com/XiaoMi/soar/cmd/soar/soar.go:264 +0xc18

qbguan@qbguan:~$ soar --version
Version: 2018-10-20 00:00:28 +0800 @7b6462c
Branch: master
Compile: 2018-10-20 00:05:33 +0800 by go version go1.10.4 linux/amd64
GitDirty: 10

--- a/ast/meta.go
+++ b/ast/meta.go
@@ -95,7 +95,7 @@ func GetMeta(stmt sqlparser.Statement, meta common.Meta) common.Meta {
 					meta[db2] = common.NewDB(db2)
 				}
 
-				meta[db1].Table[tb2] = common.NewTable(tb2)
+				meta[db2].Table[tb2] = common.NewTable(tb2)
 			}
 
 			return false, nil

--- a/ast/testdata/TestPretty.golden
+++ b/ast/testdata/TestPretty.golden
@@ -630,6 +630,8 @@ ALTER TABLE t1 CHANGE b a INT NOT NULL;
 
 ALTER TABLE  
   t1  CHANGE  b  a  INT  NOT  NULL;
+create table hello.t (id int unsigned);
+create  table  hello. t  (id  int  unsigned);
 SELECT * FROM film WHERE length = 86;
 
 SELECT  

--- a/common/cases.go
+++ b/common/cases.go
@@ -23,6 +23,9 @@ func init() {
 	// 所有的SQL都要以分号结尾，-list-test-sqls参数会打印这个list，以分号结尾可方便测试
 	// 如：./soar -list-test-sql | ./soar
 	TestSQLs = []string{
+		// DDL
+		"create table hello.t (id int unsigned);",
+
 		//  single equality
 		"SELECT * FROM film WHERE length = 86;",    // index(length)
 		"SELECT * FROM film WHERE length IS NULL;", // index(length)

